Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -707,25 +742,26 @@ protected Sheet createSheet() {
                         file == null ? "" : StringUtils.defaultString(file.getMd5Hash())));
             }
         } else {
-            String dataSourceStr = "";
-            try {
-                Content dataSource = srcContent.getDataSource();
-                if (dataSource != null) {
-                    dataSourceStr = dataSource.getName();
-                } else {
-                    dataSourceStr = getRootAncestorName();
-                }
-            } catch (TskCoreException ex) {
-                logger.log(Level.SEVERE, MessageFormat.format("Error getting source data source name (artifact objID={0})", artifact.getId()), ex); //NON-NLS
-
-            }
+            if(dataSourceName == null) {
+                try {
+                    Content dataSource = srcContent.getDataSource();
```

Comment:
I'd remove this block.  We already tried to make the name in the key. If that failed, then we just display nothing.